### DISCUSSION TITLE
If I want to start model training from scratch

### DISCUSSION THREAD

**Discussion by junbon1120** [Equim-chan/Mortal#57] [Q&A]
Hi, I had a difficulty when I tried to start training a model using the Mortal source code.
When I try to run "train.py", it loads the trained model from the config file for the "test_player" and starts training the model.
I would like a clue as to what to do if I don't have a trained model for "test_player".

**Comment by Equim-chan** [comment]
Currently the code doesn't handle bootstrapping yet, but I think you could just temporarily comment out the load-from-file part in `TestPlayer`, which will keep the params randomized, until you have a model that can be used as a baseline.

**Reply by junbon1120** [reply]
Thanks for the reply.

As you pointed out, I commented out the part about Testplayer.py and ran the code. In that case, config['grp']['state_file'] needs to be read by FileDatasetsIter() on line 183, but state_file does not exist because I want to start model training from scratch.

Is it appropriate to comment it out in this case as well? I am wondering if I need to run it from train_grp.py first, what are your thoughts

**Reply by Equim-chan** [reply]
Yes, a GRP model is required to train Mortal.

**Reply by junbon1120** [reply]
I am currently in the process of preparing training data from Tenhou logs to run train_grp.py.
My question is, am I right in thinking that the training data is JSON files compressed in GZip format, not mjai format?
I would appreciate it if you could tell me which code in Git Hub to use to annotate the training data from Tenhou logs.

**Reply by Equim-chan** [reply]
Mortal only recognizes mjai format which encodes to newline-separated JSON. GZip is optional.

**Reply by junbon1120** [reply]
Do you use libriichi to convert Tenhou log data to mjai format?

**Reply by Equim-chan** [reply]
No. Mortal and libriichi only recognize mjai.

**Reply by junbon1120** [reply]
Thanks, I will try to convert Tenhou log data to mjai format using mjai-reviewer code!

**Reply by junbon1120** [reply]
Hai
I was able to convert to mjai format using the mjai-reviewer code. Thanks for your comment.
I am aware that I need to run "train_grp.py" and "train.py" to learn Mortal, is this understanding correct?
Do I need to run any other Code?

**Reply by HALF111** [reply]
@junbon1120 Hi! I'm sorry to bother you! I also encountered the same problems as you, including (1) how to convert Tenhou log data to mjai format; and (2) whether simply running "train_grp.py" and "train.py" to learn Mortal is enough. I don't know if you have successfully solved these problems.

**Comment by jacy** [comment]
what is the project for if can't train from scratch?

**Comment by phu54321** [comment]
How much training time is spent on the current online model?

**Comment by phu54321** [comment]
Since this whole project is under AGPL, should I distribute the weights if I plan to create an web app based on this? Thanks

**Reply by Equim-chan** [reply]
You don't have to, since weights are not a part of the source code.
